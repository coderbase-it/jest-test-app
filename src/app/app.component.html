

<app-button buttonTitle="super title" (buttonClicked)='goToNextPage()'></app-button>
<app-table [headers]="['col1']">
  <tr>
    <th scope="row">1</th>
    <td>Mark</td>
    <td>Otto</td>
    <td>@mdo</td>
  </tr>
</app-table>
<h1>Hello</h1>
<router-outlet></router-outlet>
