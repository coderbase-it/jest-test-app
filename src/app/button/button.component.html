<button *ngIf="buttonTitle" (click)="clickOnButton()">
  {{ buttonTitle}}
</button>
