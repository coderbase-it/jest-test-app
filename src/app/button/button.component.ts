import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';


@Component({
  selector: 'app-button',
  templateUrl: './button.component.html',
  styleUrls: ['./button.component.scss']
})
export class ButtonComponent implements OnInit {
  @Input() buttonTitle: string;
  @Output() buttonClicked = new EventEmitter()

  private privateProperty = false;
  constructor() { }

  ngOnInit(): void {
  }

  clickOnButton(){
    //console.log('hello')
    this.buttonClicked.emit()
  }

  private privateMethod(){
    return false
  }
}
