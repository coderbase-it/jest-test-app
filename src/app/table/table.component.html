<table class="table table-dark">
  <thead>
    <tr>
      <th scope="col" *ngFor="let header of headers">{{header}}</th>
    </tr>
  </thead>
  <tbody>
   <ng-content></ng-content>
  </tbody>
</table>
