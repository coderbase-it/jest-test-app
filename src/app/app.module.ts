import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { ButtonComponent } from './button/button.component';
import { TableComponent } from './table/table.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { HttpClientModule} from '@angular/common/http';
@NgModule({
  declarations: [
    AppComponent,
    ButtonComponent,
    TableComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    NgbModule,
    HttpClientModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
