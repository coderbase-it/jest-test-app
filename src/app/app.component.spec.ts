import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { RouterOutlet } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { ButtonComponent } from './button/button.component';
import { TableComponent } from './table/table.component';


/**
 * ISOLATED TESTING EXAMPLE
 */

describe('App component test suite', () => {
  let component: AppComponent;
  beforeEach(  () => {
     component = new AppComponent()
  })

  it('my component is working', () => {
    //const component = new AppComponent()
    expect(component).toBeTruthy()

  })

 // it.todo('check that method gotoNextPage exist ')
 it('check that method gotoNextPage exist ', ()=> {
  //const component = new AppComponent()
  expect(component.goToNextPage).toBeDefined();
 })


 it('should have a title with string jest-test-app  ', ()=> {
  //const component = new AppComponent()
  expect(component.title).toEqual('jest-test-app')
 })
})

/**
 * Shallow testing
 */


describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
       // RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
      schemas: [ NO_ERRORS_SCHEMA] // this is link with shallow testing
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'jest-test-app'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('jest-test-app');
  });

  it('Check template  contain h1 and app-button tag ', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement;

    expect(compiled.querySelector('h1')).toBeTruthy()
    expect(compiled.querySelector('app-button')).toBeTruthy()
    //expect(compiled.querySelector('.content span').textContent).toContain('jest-test-app app is running!');
  });
});


/**
 * Integrated Testing
 */


 describe('AppComponent', () => {
   let fixture: ComponentFixture<AppComponent>
   let component : AppComponent;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
       RouterTestingModule
      ],
      declarations: [
        AppComponent, ButtonComponent, TableComponent
      ]
    }).compileComponents();
  });
  beforeEach(( ) => {
    fixture = TestBed.createComponent(AppComponent);
    component  =  fixture.componentInstance;
    fixture.detectChanges();
  })

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'jest-test-app'`, () => {

    expect(component.title).toEqual('jest-test-app');
  });

  it(`should have router-outlet'`, () => {
    const routeroutLet = fixture.debugElement.query(By.directive(RouterOutlet));

		expect(routeroutLet).toBeDefined()
  });

  it('Check template  contain h1 and app-button tag ', () => {
    const compiled = fixture.nativeElement;
    console.log(compiled.innerHTML)
    // expect(compiled.querySelector('h1')).toBeTruthy()
    // expect(compiled.querySelector('app-button')).toBeTruthy()
    //expect(compiled.querySelector('.content span').textContent).toContain('jest-test-app app is running!');
  });
});
