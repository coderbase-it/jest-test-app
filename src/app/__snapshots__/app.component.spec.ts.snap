// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot testing 1`] = `
<app-root
  __ngContext__={[Function LRootView]}
  title={[Function String]}
>
  <app-button
    buttontitle="super title"
  /><app-table>
    <tr>
      <th
        scope="row"
      >
        1
      </th>
      <td>
        Mark
      </td>
      <td>
        Otto
      </td>
      <td>
        @mdo
      </td>
    </tr>
  </app-table><h1>
    Hello
  </h1><router-outlet />
</app-root>
`;

exports[`AppComponent snapshot testing 2`] = `
<app-root
  __ngContext__={[Function LRootView]}
  title={[Function String]}
>
  <app-button
    buttontitle="super title"
  >
    <button>
       super title
  
    </button>
    
  </app-button><app-table>
    <table
      class="table table-dark"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            col1
          </th>
          
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Mark
          </td>
          <td>
            Otto
          </td>
          <td>
            @mdo
          </td>
        </tr>
      </tbody>
    </table>
  </app-table><h1>
    Hello
  </h1><router-outlet />
</app-root>
`;
